MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -2388,24 +2388,27 @@
 	_metalFeatures.mtlBufferAlignment = 256;	// Even on Apple Silicon
 #endif
 
-	// Currently, Metal argument buffer support is in beta stage, and is only supported
-	// on macOS 11.0 (Big Sur) or later, or on older versions of macOS using an Intel GPU.
-	// Metal argument buffers support is not available on iOS. Development to support iOS
-	// and a wider combination of GPU's on older macOS versions is under way.
-#if MVK_MACOS
-	_metalFeatures.descriptorSetArgumentBuffers = (_metalFeatures.argumentBuffers &&
-												   (mvkOSVersionIsAtLeast(11.0) ||
-													_properties.vendorID == kIntelVendorId));
-#endif
-	// Currently, if we don't support descriptor set argument buffers, we can't support argument buffers.
-	_metalFeatures.argumentBuffers = _metalFeatures.descriptorSetArgumentBuffers;
-
+	// Argument buffers
 	if ([_mtlDevice respondsToSelector: @selector(argumentBuffersSupport)]) {
 		_metalFeatures.argumentBuffersTier = _mtlDevice.argumentBuffersSupport;
 	} else {
 		_metalFeatures.argumentBuffersTier = MTLArgumentBuffersTier1;
 	}
 
+	// Metal argument buffer support for descriptor sets is supported on macOS 11.0 or later,
+	// or on older versions of macOS using an Intel GPU, or on iOS & tvOS 16.0 or later (Metal 3).
+	_metalFeatures.descriptorSetArgumentBuffers = (_metalFeatures.argumentBuffers &&
+												   (mvkOSVersionIsAtLeast(11.0, 16.0, 1.0) ||
+													_properties.vendorID == kIntelVendorId));
```

Comment:
Something I need to do is, test argument buffers on my Mac Pro with AMD hardware.